src/Microsoft.AspNetCore.Blazor.Browser/Interop/JavaScriptInvoke.cs

```diff
@@ -0,0 +1,253 @@
+                                        .SingleOrDefault(m => m.Name == "ResultOnSuccess") ??
+                                    task.GetType()
+                                        .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                                        .SingleOrDefault(m => m.Name == "Result");
+
+                                var returnValue = resultProperty.GetValue(task);
+                                RegisteredFunction.Invoke<bool>(
+                                    options.Async.FunctionName,
+                                    options.Async.ResolveId,
+                                    returnValue);
+                            }
+                        }
+                        else
+                        {
+                            RegisteredFunction.Invoke<bool>(
+                                options.Async.FunctionName,
+                                options.Async.RejectId);
+                        }
+
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e.ToString());
+                        throw;
+                    }
+                    finally
+                    {
+                        Console.WriteLine("Finally!");
+                    }
+                });
+            }
+
+            return JsonUtil.Serialize(result);
+        }
+
+        internal static MethodInfo ResolveMethod(MethodOptions options)
+        {
+            throw new NotImplementedException();
+        }
+
+        private static Func<string, object> GetOrCreateInvoker(MethodOptions options, Func<string, object[]> argumentDeserializer)
+        {
+            var method = options.GetMethodOrThrow();
+            return (string args) => method.Invoke(null, argumentDeserializer(args));
+        }
+
+        private static Func<string, object[]> GetOrCreateArgumentDeserializer(MethodOptions options)
+        {
+            var info = options.GetMethodOrThrow();
+            var argsClass = ArgumentList.GetArgumentClass(info.GetParameters().Select(p => p.ParameterType).ToArray());
+            var deserializeMethod = ArgumentList.GetDeserializer(argsClass);
+            var toParameterListMethod = argsClass.GetMethods(BindingFlags.Instance | BindingFlags.Public)
+                .Where(m => string.Equals(nameof(ArgumentList.ToParameterList), m.Name))
+                .Single();
+
+            return Deserialize;
+
+            object[] Deserialize(string arguments)
+            {
+                var argsInstance = deserializeMethod(arguments);
+                return argsInstance.ToParameterList();
+            }
+        }
+
+        /// <summary>
+        /// TODO
+        /// </summary>
+        /// <param name="identifier"></param>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        public static Task InvokeJavaScriptVoidFunctionAsync(string identifier, params object[] args)
```

Comment:
Could the name for this be `InvokeJavaScriptFunctionAsync`? i.e., both overloads could have the same name, differentiated only by the presence or absence of a generic arg. Might be easier for callers to understand.

Reply:
This goes away